lib/internal/Magento/Framework/Validator/Factory.php

```diff
@@ -6,65 +6,50 @@
 
 namespace Magento\Framework\Validator;
 
-use Magento\Framework\Cache\FrontendInterface;
+use Magento\Framework\Module\Dir\Reader;
+use Magento\Framework\ObjectManagerInterface;
+use Magento\Framework\Phrase;
+use Magento\Framework\Validator;
 
+/**
+ * Factory for \Magento\Framework\Validator and \Magento\Framework\Validator\Builder.
+ */
 class Factory
 {
-    /** cache key */
-    const CACHE_KEY = __CLASS__;
-
     /**
-     * @var \Magento\Framework\ObjectManagerInterface
+     * @var ObjectManagerInterface
      */
     protected $_objectManager;
 
     /**
      * Validator config files
      *
-     * @var array|null
+     * @var iterable|null
      */
-    protected $_configFiles = null;
+    protected $_configFiles;
 
     /**
      * @var bool
      */
     private $isDefaultTranslatorInitialized = false;
 
     /**
-     * @var \Magento\Framework\Module\Dir\Reader
+     * @var Reader
      */
     private $moduleReader;
 
-    /**
-     * @var FrontendInterface
-     */
-    private $cache;
-
-    /**
-     * @var \Magento\Framework\Serialize\SerializerInterface
-     */
-    private $serializer;
-
-    /**
-     * @var \Magento\Framework\Config\FileIteratorFactory
-     */
-    private $fileIteratorFactory;
-
     /**
      * Initialize dependencies
      *
-     * @param \Magento\Framework\ObjectManagerInterface $objectManager
-     * @param \Magento\Framework\Module\Dir\Reader $moduleReader
-     * @param FrontendInterface $cache
+     * @param ObjectManagerInterface $objectManager
+     * @param Reader $moduleReader
      */
     public function __construct(
-        \Magento\Framework\ObjectManagerInterface $objectManager,
-        \Magento\Framework\Module\Dir\Reader $moduleReader,
-        FrontendInterface $cache
```

Comment:
Please, leave FrontendInterface as constructor parameter and mark it as deprecated, otherwise it's backward incompatible change

Reply:
Thanks for your feedback. I addressed this in my latest commit.